File: scm-java/src/main/scheme/primitives/PrimitiveCharNumericP.java
package scheme.primitives;

import scheme.Primitive;
import scheme.Value;
import scheme.SourcePos;

public class PrimitiveCharNumericP extends Primitive {
    @Override
    public String name() {
        return "char-numeric?";
    }

    @Override
    public String info() {
        return "Syntax: (char-numeric? char)\n" +
               "Library: (scheme char)\n" +
               "Description: Returns #t if char is a numeric character (digit).\n" +
               "Example:\n" +
               "  (char-numeric? #\\5) => #t\n" +
               "  (char-numeric? #\\a) => #f";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return Character.isDigit(Value.asChar(arguments[0]));
    }
}

File: scm-csharp/scheme/primitives/PrimitiveCharNumericP.cs
namespace scheme;

public class PrimitiveCharNumericP : Primitive
{
    public override string Name()
    {
        return "char-numeric?";
    }

    public override string Info()
    {
        return
            "Syntax: (char-numeric? char)\n" +
            "Library: (scheme char)\n" +
            "Description: Returns #t if char is a numeric character (digit).\n" +
            "Example:\n" +
            "  (char-numeric? #\\5) => #t\n" +
            "  (char-numeric? #\\a) => #f";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return Char.IsDigit(Value.AsChar(arguments[0]));
    }
}
